### PR TITLE
Rewrite specification

### PR DESCRIPTION
**Pull request by wooorm**
* Reword `text` > `literal` (GH-17)
* Add introduction
* Reword definitions, terms
* Add more terms
* Add references
* Add more acknowledgements

Closes GH-17.

### CODE REVIEW

**Comment by wooorm** [comment]
Looking for reviewers! 👓

**Comment by wooorm** [comment]
/CC @ChristianMurphy @ikatyang perchance?

Feel free to ignore, I’ll merge in a few days.

File: readme.md [hunk 1]
```diff
@@ -1,61 +1,91 @@
...
-### `Node`
+Syntax trees are representations of source code or even natural language.
+These trees are abstractions that make it possible to analyse, transform,
+and generate code.
+
+Syntax trees come in two flavours:
+
+*   **concrete syntax trees**: structures that represent every detail (such
+    as white-space in white-space insensitive languages)
+*   **abstract syntax trees**: structures that only represent details relating
```
Comment (ChristianMurphy) [inline readme.md]: Maybe link out to https://en.m.wikipedia.org/wiki/Abstract_syntax_tree ?

Comment (wooorm) [reply readme.md]: That one’s good, but I find the [related CST article](https://en.m.wikipedia.org/wiki/Parse_tree) a bit lacking! I also really like this article: https://eli.thegreenplace.net/2009/02/16/abstract-vs-concrete-syntax-trees/.

Comment (wooorm) [reply readme.md]: Done!

**Pull request closed without merging**